types/serverconfig.pp

```diff
@@ -0,0 +1,11 @@
+type Puppetwebhook::Serverconfig = Struct[{
+  server_type => Enum['simple', 'daemon'],
+  logfile     => Stdlib::Absolutepath,
+  loglevel    => Enum['WARN', 'INFO', 'DEBUG', 'ERROR'],
+  pidfile     => Stdlib::Absolutepath,
+  port        => Integer,
+  enable_ssl  => Boolean,
+  ssl_verify  => Boolean,
+  ssl_cert    => Variant[String, Stdlib::Absolutepath],
```

Comment:
An absolute path is a string so I'd use that. In case you want to say empty string I'd use `Enum['']` instead.